pkg/lib/facade/coordinator.go

```diff
@@ -417,6 +426,127 @@ func (c *Coordinator) AuthenticatorVerifyOneWithSpec(userID string, authenticato
 	return
 }
 
+func (c *Coordinator) UserCreatebyAdmin(identitySpec *identity.Spec, password string) (*user.User, error) {
+	// 1. Create user
+	userID := uuid.New()
+	user, err := c.UserCommands.Create(userID)
+	if err != nil {
+		return nil, err
+	}
+
+	// 2. Create identity
+	identityInfo, err := c.Identities.New(userID, identitySpec, identity.NewIdentityOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := c.Identities.CheckDuplicated(identityInfo); err != nil {
+		return nil, err
+	}
+
+	if err := c.Identities.Create(identityInfo); err != nil {
+		return nil, err
+	}
+
+	if err := c.StdAttrsService.PopulateIdentityAwareStandardAttributes(userID); err != nil {
+		return nil, err
+	}
+
+	// 3. Create primary authenticators
+	authenticatorInfos, err := c.createPrimaryAuthenticators(identityInfo, userID, password)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(authenticatorInfos) == 0 {
+		return nil, api.InvalidConfiguration.New("no primary authenticator can be created for identity")
+	}
+
+	if err := c.UserCommands.AfterCreate(
+		user,
+		[]*identity.Info{identityInfo},
+		authenticatorInfos,
+		true,
+	); err != nil {
+		return nil, err
+	}
+
+	return user, nil
+}
+
+func (c *Coordinator) createPrimaryAuthenticators(identityInfo *identity.Info, userID string, password string) ([]*authenticator.Info, error) {
+	authenticatorTypes := *c.AuthenticationConfig.PrimaryAuthenticators
+	if len(authenticatorTypes) == 0 {
+		return nil, api.InvalidConfiguration.New("identity requires primary authenticator but none is enabled")
+	}
+
+	var authenticatorSpecs []*authenticator.Spec
+	for _, t := range authenticatorTypes {
+		switch t {
+		case model.AuthenticatorTypePassword:
+			if len(password) == 0 {
+				return nil, api.NewInvariantViolated(
+					"PasswordRequired",
+					"password is required",
+					nil,
+				)
+			}
```

Comment:
This is incorrect. Password is not required when we can create other authenticator.